### PR TITLE
Implement patch ops for branches and NOP

### PR DESCRIPTION
**Pull request by KarlVogel**
Adds functions to the arch to enable the context sensitive patch menu in BN to allow patching the conditional branches. The patch "always" is a bit messy though.

### CODE REVIEW

**Comment by fluxchief** [comment]
Thanks a lot for the PR!
`perform_is_never_branch_patch_available` and `perform_is_always_branch_patch_available` implement exactly the same behavior - I think it would be useful to add a function `is_branching_instruction` or similar that will be used in both cases.
I'll have a look later at the branch always and if there is a cleaner way to implement this. I'd at least use struct.pack in favor of chr(), what do you think?

**Comment by KarlVogel** [comment]
Sounds good to me. Was first thinking of using isinstance of Instruction_BR_Abstract to do the check for branch instruction, but noticed perform_get_instruction_info was listing the classes, so re-used that code bit. Since it's repeated 3 times in the source, it makes sense to put it in a separate function.
Will rework the patch later today.

**Comment by fluxchief** [comment]
Feel free to make use of isinstance `Instruction_BR_Abstract`, in this case just change `perform_get_instruction_info` as well. I think this would make it more readable.

File: __init__.py [hunk 1]
```diff
@@ -262,6 +247,28 @@ def perform_get_instruction_low_level_il(self, data, addr, il):
...
+        v = (v & 0xFFF) | 0xc000
+        return struct.pack('<H', v)
+
+    def perform_never_branch(self, data, addr):
+        return "\x00\x00"
+
+    def perform_convert_to_nop(self, data, addr):
+        return "\x00\x00"
+
+
```
Comment (fluxchief) [inline __init__.py]: One last thing, remove this additional newline here, then we should be good :+1:

**Comment by fluxchief** [comment]
I added one more comment, then we're good :)

**Comment by KarlVogel** [comment]
I blame my keyboard for that one! :-)

**Comment by fluxchief** [comment]
Thanks!

**Pull request merged**